src/mame/drivers/rainbow.cpp

```diff
@@ -754,20 +754,19 @@ UPD7220_DISPLAY_PIXELS_MEMBER( rainbow_state::hgdc_display_pixels )
 		return; // no output from graphics option
 	}
 
-	address = ( m_GDC_SCROLL_BUFFER[ ((address & 0x7FC0) >> 7) & 0xff ] << 7) |  (address & 0x7F);
-
 	// ********************* GET BITMAP DATA FOR 4 PLANES ***************************************
 	// _READ_ BIT MAP  from 2 or 4 planes (plane 0 is least, plane 3 most significant). See page 42 / 43
 	if(m_GDC_MODE_REGISTER & GDC_MODE_HIGHRES)
 	{
+			address = ( m_GDC_SCROLL_BUFFER[ ((address & 0x7FC0) >> 7) & 0xff ] << 7) |  (address & 0x7F);
 			plane0 = m_video_ram[((address & 0x7fff) + 0x00000) >> 1];
 			plane1 = m_video_ram[((address & 0x7fff) + 0x10000) >> 1];
 			plane2 = plane3 = 0;
 	}
 		else
```

Comment:
Please line this else statement up properly while you're editing code in the area